etcdctlv3/command/snapshot_command.go

```diff
@@ -88,13 +90,39 @@ func snapshotToFile(c *clientv3.Client, path string) {
 // snapshot reads all of a watcher; returns compaction revision if incomplete
 // TODO: stabilize snapshot format
 func snapshot(w io.Writer, c *clientv3.Client, rev int64) int64 {
-	wapi := clientv3.NewWatcher(c)
-	defer wapi.Close()
+	s := sync.NewSyncer(c, "", rev)
+
+	rc, errc := s.SyncBase(context.TODO())
+
+	more := true
+	for more {
+		select {
+		case err, ok := <-errc:
```

Comment:
bug: rc buffers a bunch of messages, state slurp completes, errc closes, `return 0` but there are still keys sitting in rc